### PR TITLE
[Quest]Implement Chips quest

### PR DESCRIPTION
**Pull request by hooksta4**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Implements Bastok quest Chips. 

## Steps to test these changes
!addmission 6 638
speak with Cid to get mission status to 1 or !setplayervar Mission[6][638]Status 1
Speak with Ghebi in Tenshodo - will return option. 
If player selects no, they must zone as per [capture](https://www.youtube.com/watch?v=CO5XbPSuY0g)
Return to Cid and trade Carmine Chip, Cyan Chip and Gray Chip to get CCB Polymer. 
NOTE if you drop it for testing, you will need to zone due to recycle bin.
Upon first completion, player will complete quest. Further trades of all 3 chips will reward another CCB Polymer.

### CODE REVIEW

File: scripts/quests/bastok/Chips.lua [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        {
+            ['Cid'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        npcUtil.tradeHasExactly(trade, {
+                        xi.item.CARMINE_CHIP,
+                        xi.item.CYAN_CHIP,
+                        xi.item.GRAY_CHIP
+                        }) and
```
Comment (Xaver-DaRed) [inline scripts/quests/bastok/Chips.lua:97]: This trade can all be 1 line

File: scripts/quests/bastok/Chips.lua [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        [xi.zone.LOWER_JEUNO] =
+        {
+            ['Ghebi_Damomohe'] = quest:progressEvent(169),
+
+            onEventFinish =
+            {
+                [169] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:begin(player)
+                    else quest:setMustZone(player)
```
Comment (claywar) [inline scripts/quests/bastok/Chips.lua:36]: formatting

File: scripts/quests/bastok/Chips.lua [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+            (player:getCurrentMission(xi.mission.log_id.COP) > xi.mission.id.cop.ONE_TO_BE_FEARED or
+            (player:getCurrentMission(xi.mission.log_id.COP) == xi.mission.id.cop.ONE_TO_BE_FEARED and
+            player:getCharVar('Mission[6][638]Status') >= 1)) and
```
Comment (claywar) [inline scripts/quests/bastok/Chips.lua:23]:
```
xi.mission.getVar = function(player, areaId, missionId, name)
    return player:getVar(getVarPrefix(areaId, missionId) .. name)
end
```

File: scripts/quests/bastok/Chips.lua [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        end,
+
+        [xi.zone.METALWORKS] =
+        {
+            ['Cid'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        npcUtil.tradeHasExactly(trade, {
+                        xi.item.CARMINE_CHIP,
```
Comment (claywar) [inline scripts/quests/bastok/Chips.lua:55]: indent needed for subparams on multiple lines, an alternative would be to define the table elsewhere

File: scripts/quests/bastok/Chips.lua [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                        xi.item.CYAN_CHIP,
+                        xi.item.GRAY_CHIP
+                        }) and
+                        not player:hasItem(xi.item.CCB_POLYMER_PUMP)
+                    then
+                        return quest:progressEvent(883)
+                    elseif
```
Comment (claywar) [inline scripts/quests/bastok/Chips.lua:62]: Is the player really limited to only one pump? Its ex, not rare

Comment (hooksta4) [reply scripts/quests/bastok/Chips.lua:62]: image
It appears from capture, that is the case.

Comment (WinterSolstice8) [reply scripts/quests/bastok/Chips.lua:62]: bizarre.

Comment (hooksta4) [reply scripts/quests/bastok/Chips.lua:62]: Yea I thought the same thing had to double check the capture at first - anyways thanks for the feedback. Made alterations. <3

**Comment by hooksta4** [comment]
Thanks, made changes.

**Comment by hooksta4** [comment]
This went sideways somehow - will close and redo it.

**Pull request closed without merging**